src/routes/items.js

```diff
@@ -32,28 +33,42 @@ router.get("/:id", async (req, res, next) => {
   }
 });
 
-router.post("/", authenticate, async (req, res, next) => {
-  const { name, price, publishYear, author, description, owner } = req.body;
-  const item = new Item({
-    _id: mongoose.Types.ObjectId(),
-    name,
-    publishYear,
-    price,
-    author,
-    description,
-    owner
-  });
-  console.log(userId);
-  try {
-    const result = await item.save();
-    res.status(201).json({
-      ok: true,
-      item: result
+router.post(
+  "/",
+  upload.single("image"),
+  authenticate,
```

Comment:
first `upload.single("image")` and then `authenticate` middle-ware?

Reply:
that was a mistake refactored a bit more check them